cpp/include/cudf/detail/sizes_to_offsets_iterator.cuh

```diff
@@ -255,12 +255,14 @@ static sizes_to_offsets_iterator<ScanIterator, LastType> make_sizes_to_offsets_i
  * @param begin Input iterator for scan
  * @param end End of the input iterator
  * @param result Output iterator for scan result
+ * @param initial_offset Initial offset to add to scan
  * @return The last element of the scan
  */
 template <typename SizesIterator, typename OffsetsIterator>
 auto sizes_to_offsets(SizesIterator begin,
                       SizesIterator end,
                       OffsetsIterator result,
+                      uint64_t initial_offset,
```

Comment:
@davidwendt modifying this utility to take in an initial offset as well.

Reply:
The function takes iterators, so the initial offset value could be incorporated into a transform iterator.
Also, the offsets are signed integers.
The return is intended to convey the total size of the scan so now the caller must subtract the `initial_offset`.
This makes the parameter as implemented here more desirable than a transform iterator I think.
I would rather it be signed `int64_t`. The `LastType` is an internal implementation detail that probably should be re-evaluated in the future.

Reply:
> I would rather it be signed int64_t.

+1

Reply:
> Also, the offsets are signed integers.
> The return is intended to convey the total size of the scan so now the caller must subtract the initial_offset.

Thank you for clarifying this. I wasn't too sure by looking at the `LastType` so chose the largest possible one.

> This makes the parameter as implemented here more desirable than a transform iterator I think.
Sounds good, let me make the initial_offset as `int64_t`.